src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -89,6 +96,7 @@ def detect() -> bool:
         This will then avoid the detection of ``SLURMEnvironment`` and another environment can be detected
         automatically.
         """
+        SLURMEnvironment._validate_srun_used()
```

Comment:
isn't it sufficient to only run this on the environment's init? Why would you want to run it also on detection? If this is used in the end, it will run in init anyways and otherwise it doesn't make sense to validate IMO

Reply:
No, I think not. That would render the check useless. The SLURM environment variables are only properly set if srun is used. The dependencies are:
srun used -> environment variables set -> detect() returns True

So, when in the AcceleratorConnector we do 

```
for env in environments:
    env.detect()
```

we want that detect() checks if srun is available but not used. If it is available and not used, a different environment gets selected because SLURMEnvironment.detect() = False and the process hangs, but the warning gets emitted and the user can see it. 

On the other hand, we also call the check in init in case user explicitly instantiates and passes the plugin.